llvm/include/llvm/IR/VPIntrinsics.def

```diff
@@ -702,6 +702,12 @@ BEGIN_REGISTER_VP(experimental_vp_splice, 3, 5, EXPERIMENTAL_VP_SPLICE, -1)
 VP_PROPERTY_FUNCTIONAL_INTRINSIC(experimental_vector_splice)
 END_REGISTER_VP(experimental_vp_splice, EXPERIMENTAL_VP_SPLICE)
 
+// llvm.experimental.vp.reverse(x,mask,vlen)
+BEGIN_REGISTER_VP(experimental_vp_reverse, 1, 2,
+                  EXPERIMENTAL_VP_REVERSE, -1)
+VP_PROPERTY_FUNCTIONAL_INTRINSIC(experimental_vector_reverse)
```

Comment:
The expand vector predication pass will expand out llvm.experimental.vp.reverse to llvm.experimental.vector.reverse and drop the EVL if we mark it as functionally equivalent, but without the EVL it won't be the same reverse. Should this be marked as `VP_PROPERTY_NO_FUNCTIONAL` instead?

Reply:
Does a similar issue exist for vp.splice, or does expand vector predication do something special?

Reply:
Yeah good catch, I'll post a PR to fix that